File: res/Modules_3_5_7_Project Resources (Copy)/python_examples/employee/java/TestEmployee.java
public class TestEmployee
{

	public static void main(String[] args)
	{
		Employee e1 = null;
		Employee e2 = null;
		try
		{
			e1 = new SalariedEmployee("John Doe",  "012345678",	new Date(12, 12, 1976), 30000);
			e2 = new HourlyEmployee("Billy Bob Hogg", "000000000", 	new Date(4, 1, 1942), 6.15, 8);
		}
		catch (InvalidSalaryException e)
		{
			System.out.println (e.getMessage());
		}
		catch (InvalidHourlyRateException e)
		{
			System.out.println (e.getMessage());
		}
		catch (InvalidHoursException e)
		{
			System.out.println (e.getMessage());
		}
		printEmployee (e1);
		printEmployee (e2);
	}
	
	public static void printEmployee (Employee e)
	{
		System.out.println (e.getName() + ": "  + e.weeklyPay());
	}
}

File: res/Modules_3_5_7_Project Resources (Copy)/python_examples/employee/python/TestEmployee.py
'''
Created on Jun 27, 2011

@author: dgayler
'''

import datetime
from SalariedEmployees import SalariedEmployee
from HourlyEmployees import HourlyEmployee
from InvalidSalaryExceptions import InvalidSalaryException
from InvalidHourlyRateExceptions import InvalidHourlyRateException
from InvalidHoursExceptions import InvalidHoursException

def printEmployee (e):
    if e is None:
        raise ValueError ("invalid employee argument")
    print (e.name + ": " + repr(e.weeklyPay()))
        
if __name__ == '__main__':
    try:
        e1 = SalariedEmployee("John Doe",  "012345678", datetime.date(1976, 12, 12), 30000)
        e2 = HourlyEmployee("Billy Bob Hogg", "000000000", datetime.date(1942, 4, 1), 6.15, 8)
        printEmployee (e1)
        printEmployee (e2)
    except InvalidSalaryException as e:
        print (e.message)
    except InvalidHourlyRateException as e:
        print (e.message)
    except InvalidHoursException as e:
        print (e.message)
    except ValueError as  e:
        print (e.message)
        
